driver-core/src/main/java/com/datastax/driver/core/ArrayBackedRow.java

```diff
@@ -239,6 +242,23 @@ public InetAddress getInet(String name) {
         return getInet(metadata.getFirstIdx(name));
     }
 
+    public Token getToken(int i) {
+        if (tokenFactory == null)
+            throw new DriverInternalError("Token factory not set. This should only happen at initialization time");
+
+        metadata.checkType(i, tokenFactory.getTokenType().getName());
+
+        ByteBuffer value = data.get(i);
+        if (value == null || value.remaining() == 0)
+            return null;
+
+        return tokenFactory.deserialize(value);
+    }
+
+    public Token getToken(String name) {
+        return getToken(metadata.getFirstIdx(name));
```

Comment:
> As an alternative to the magic constant, we could have dedicated methods

Yes, that's probably even cleaner.

> Or even simpler, `getToken()` and `setToken(Token)`

I think I prefer `getPartitionKeyToken()/setPartitionKeyToken()` as it's more explicit what they do get/set. I'll admit that it's likely most people won't get confused if we use the order name, because again there isn't much use to get/set the token of anything else than the parititon key, but the more explicit naming could avoid confusing a couple users, and to some extend it's a better documentation of what the method does. I don't feel terribly strongly though.

> I think it's worth leaving the "i => token(i)" logic in `getToken(String)`.

Again, I worry that the inconsistency with {{setToken}} will surprise people, and I'd prefer limiting the magic to as many places as possible. If we do add `getPartitionKeyToken()/setPartitionKeyToken()`, then the only valid reason to even have `getToken(String)/setToken(String, Token)` is for when the user uses an alias (`getToken`) or a named bind marker (`setToken`). So I'd be in favor of reverting `getToken(String)/setToken(String, Token)` to their initial behavior (which is simple and consistent) and be very explicit in their javadoc that you should favor `getPartitionKeyToken()/setPartitionKeyToken()` unless you're using an alias/named bind marker.